app/models/king.rb

```diff
@@ -1,5 +1,48 @@
 class King < Piece
 
+  def valid_move?(new_x, new_y)
+    #checks if the move that the king is making is valid
+    if position_exist?(new_x, new_y) && pass_king_rules?(new_x, new_y)
+      return true
+    else
+      return false
+    end
```

Comment:
I think this is equivalent to

``` ruby
def valid_move?(new_x, new_y)
  position_exist?(new_x, new_y) && pass_king_rules?(new_x, new_y)
end
```

because the && will return a boolean?